test/integration/features/generic-driver.feature

```diff
@@ -0,0 +1,49 @@
+@generic-driver
+Feature: With generic driver Minishift can provision remote unprovisioned VM 
+
+  Scenario: User creates an unprovisioned VM
+    Given setting up environment variable "MINISHIFT_ENABLE_EXPERIMENTAL" with value "y" succeeds
+    When executing "minishift profile set remote" succeeds
+    And executing "minishift start --no-provision" succeeds
+    Then Minishift should have state "Running"
+
+  Scenario: Provisioning remote instance
+    Given unset environment variable "MINISHIFT_ENABLE_EXPERIMENTAL" succeeds
+    And setting scenario variable "REMOTE_VM_IP" to the stdout from executing "minishift ip"
+    And executing "minishift profile set minishift" succeeds
+    When executing "minishift start --vm-driver generic --remote-ipaddress $(REMOTE_VM_IP) --remote-ssh-key .minishift/profiles/remote/machines/remote/id_rsa --remote-ssh-user docker" succeeds
+    Then Minishift should have state "Running"
+
+  Scenario: Stopping with generic driver
+  Stopping with generic driver puts oc cluster down, remote VM however keeps running.
+    When executing "minishift stop" succeeds
+    And executing "minishift status" succeeds
+    Then stdout should match "OpenShift:\s+Stopped"
+
+  Scenario: VM at remote location is still running
+    When executing "minishift status --profile remote" succeeds
+    Then stdout should match "Minishift:\s+Running"
+    And stdout should match "OpenShift:\s+Stopped"
+
+  Scenario: Starting stopped cluster with generic driver
```

Comment:
I don not think its starting the remote stooped openshift cluster instead its starting a cluster inside minishift profile. I am expecting the steps should be something like
```
minishift start --vm-driver generic --remote-ipaddress $(REMOTE_VM_IP) --remote-ssh-key .minishift/profiles/remote/machines/remote/id_rsa --remote-ssh-user docker" succeeds
```

Reply:
@amitkrout From my observations it does not need to repeat the flags on start from stopped. These are stored and used, this step does not create new VM locally.

Reply:
It's very hard to believe that it's not creating ```minishift``` vm. I think lets @praveenkumar  share his thoughts on the same

Reply:
I am following exactly the scenario steps except I started the VM on virtual box. Then I am executing the steps no 29 with and without vm-driver flag

Just see the difference in the output log. I am using macOS
```
$ ./minishift start
-- Starting profile 'minishift'
-- Check if deprecated options are used ... OK
-- Checking if https://github.com is reachable ... OK
-- Checking if requested OpenShift version 'v3.9.0' is valid ... OK
-- Checking if requested OpenShift version 'v3.9.0' is supported ... OK
-- Checking if requested hypervisor 'xhyve' is supported on this platform ... OK
-- Checking if xhyve driver is installed ... 
   Driver is available at /usr/local/bin/docker-machine-driver-xhyve
   Checking for setuid bit ... OK
-- Checking the ISO URL ... OK
-- Checking if provided oc flags are supported ... OK
-- Starting the OpenShift cluster using 'xhyve' hypervisor ...
-- Starting Minishift VM ........ OK
-- Checking for IP address ... OK
-- Checking for nameservers ... OK
-- Checking if external host is reachable from the Minishift VM ... 
   Pinging 8.8.8.8 ... OK
-- Checking HTTP connectivity from the VM ... 
   Retrieving http://minishift.io/index.html ... OK
-- Checking if persistent storage volume is mounted ... OK
-- Checking available disk space ... 10% used OK
-- OpenShift cluster will be configured with ...
   Version: v3.9.0
-- Copying oc binary from the OpenShift container image to VM ... OK
-- Starting OpenShift cluster .......
Using nsenter mounter for OpenShift volumes
Using public hostname IP 192.168.99.101 as the host IP
Using 192.168.99.101 as the server IP
Starting OpenShift using openshift/origin:v3.9.0 ...
OpenShift server started.

The server is accessible via web console at:
    https://192.168.99.101:8443
```
```
$ ./minishift start --vm-driver generic --remote-ipaddress 192.168.99.101 --remote-ssh-key /Users/amit/.minishift/profiles/remote/machines/remote/id_rsa --remote-ssh-user docker
-- Starting profile 'minishift'
-- Checking if provided oc flags are supported ... OK
-- Starting the OpenShift cluster using 'generic' hypervisor ...
-- Starting to provision the remote machine ........ OK
-- OpenShift cluster will be configured with ...
   Version: v3.9.0
-- Copying oc binary from the OpenShift container image to VM ... OK
-- Starting OpenShift cluster .......
Using nsenter mounter for OpenShift volumes
Using public hostname IP 192.168.99.101 as the host IP
Using 192.168.99.101 as the server IP
Starting OpenShift using openshift/origin:v3.9.0 ...
OpenShift server started.

The server is accessible via web console at:
    https://192.168.99.101:8443
```
@agajdosi I am sure you also will be seeing almost the same difference though you are provisioning the vm on default hypervisor. And more surprisingly it is starting the same openshift cluster. @praveenkumar it something we need to address? I feel this is killing the basic rules of minishift start

Reply:
@amitkrout Which basic rules of Minishift start?

AFAIU this is expected. When --vm-driver is not specified, or is kvm, xhyve, virtualbox etc... then driver checks are executed but then IP address, username and ssh key saved in machines config is used and it all goes as if generic was selected, because the machine is generic.

When --vm-driver generic is specified, then driver checks are skipped as I believe they are skipped in code around generic. In this case I think it does not matter whether you add --remote-ip-address or anything else.

Reply:
Let me put it in a general way

1. Setup virtual box on macOS or Linux so that user can switch between drivers
2. Run ```./minishift start —vm-driver virtualbox```
3. ```./minishift stop```
4. ```./mishift start```

This will run on default driver as no driver is added to the persistent config I mean to the config.json file. Ideally, this should fail if I am not wrong, however, it's not failing if I consider the same way as in case of generic driver. 

Please try out these steps so that we will be on the same page